services/venice-controller/src/main/java/com/linkedin/venice/controller/DeferredVersionSwapService.java

```diff
@@ -0,0 +1,160 @@
+ */
+public class DeferredVersionSwapService extends AbstractVeniceService {
+  private final AtomicBoolean stop = new AtomicBoolean(false);
+  private final Set<String> allClusters;
+  private final VeniceControllerMultiClusterConfig veniceControllerMultiClusterConfig;
+  private final VeniceParentHelixAdmin veniceParentHelixAdmin;
+  private final ScheduledExecutorService deferredVersionSwapExecutor = Executors.newSingleThreadScheduledExecutor();
+  private final Time time;
+  private static final Logger LOGGER = LogManager.getLogger(DeferredVersionSwapService.class);
+
+  public DeferredVersionSwapService(
+      VeniceParentHelixAdmin veniceHelixAdmin,
+      VeniceControllerMultiClusterConfig multiClusterConfig) {
+    this(veniceHelixAdmin, multiClusterConfig, new SystemTime());
+  }
+
+  protected DeferredVersionSwapService(
+      VeniceParentHelixAdmin admin,
+      VeniceControllerMultiClusterConfig multiClusterConfig,
+      Time time) {
+    this.veniceParentHelixAdmin = admin;
+    this.time = time;
+    this.allClusters = multiClusterConfig.getClusters();
+    this.veniceControllerMultiClusterConfig = multiClusterConfig;
+  }
+
+  @Override
+  public boolean startInner() throws Exception {
+    deferredVersionSwapExecutor.scheduleAtFixedRate(
+        new DeferredVersionSwapTask(),
+        0,
+        veniceControllerMultiClusterConfig.getDeferredVersionSwapSleepMs(),
+        TimeUnit.MILLISECONDS);
+    return true;
+  }
+
+  @Override
+  public void stopInner() throws Exception {
+    stop.set(true);
+    deferredVersionSwapExecutor.shutdown();
+  }
+
+  private Set<String> getRegionsForVersionSwap(Map<String, Integer> candidateRegions, Set<String> targetRegions) {
+    Set<String> remainingRegions = new HashSet<>(candidateRegions.keySet());
+    remainingRegions.removeAll(targetRegions);
+    return remainingRegions;
+  }
+
+  private int getTargetVersionFromTargetRegion(Map<String, Integer> coloToVersions, Set<String> targetRegions) {
+    int targetVersion = 0;
+    for (String targetRegion: targetRegions) {
+      targetVersion = coloToVersions.get(targetRegion);
+    }
+    return targetVersion;
+  }
+
+  private class DeferredVersionSwapTask implements Runnable {
+    @Override
+    public void run() {
+      while (!stop.get()) {
+        for (String cluster: allClusters) {
+          List<Store> stores = veniceParentHelixAdmin.getAllStores(cluster);
+          for (Store store: stores) {
+            if (StringUtils.isEmpty(store.getTargetSwapRegion())) {
+              continue;
+            }
+
+            Map<String, Integer> coloToVersions =
+                veniceParentHelixAdmin.getCurrentVersionsForMultiColos(cluster, store.getName());
+            Set<String> targetRegions = RegionUtils.parseRegionsFilterList(store.getTargetSwapRegion());
+            Set<String> remainingRegions = getRegionsForVersionSwap(coloToVersions, targetRegions);
+
+            // Do not perform version swap for davinci stores
```

Comment:
Keep a TODO comment here as later we need to remove this check once DVC delayed ingestion work is done.